users/serializers.py

```diff
@@ -0,0 +1,73 @@
+import django.contrib.auth.password_validation as password_validation
+from django.contrib.auth.models import User
+from rest_framework import serializers
+from rest_framework.authtoken.models import Token
+
+
+class UserDetailSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ('email', 'username',)
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = User
+        fields = ('email', 'username', 'password')
+
+    def create(self, data):
+        user = User(username=data["username"])
+        user.set_password(data["password"])
+
+        if "email" in data:
+            user.email = data["email"]
+
+        user.save()
+        Token.objects.create(user=user)
+        return user
+
+    def validate_password(self, value):
+        password_validation.validate_password(value)
+        return value
+
+
+class LoginSerializer(serializers.Serializer):
+    username = serializers.CharField(required=True)
+    password = serializers.CharField(required=True)
+
+    def validate(self, data):
+        try:
+            user = User.objects.get(username=data["username"])
+        except Exception as e:
+            raise serializers.ValidationError(detail={"username": e.args[0]})
+
+        if not user.check_password(data["password"]):
+            raise serializers.ValidationError(detail={"password": "Wrong password."})
+
+        token, created = Token.objects.get_or_create(user=user)
+        data["user"] = user
+        data["token"] = token.key
+        return data
+
+    def to_representation(self, data):
+        return {
+            'username': data["username"],
+            'token': data["token"]
+        }
+
+
+class ChangePasswordSerializer(serializers.Serializer):
+    old_password = serializers.CharField(required=True)
+    new_password = serializers.CharField(required=True, validators=[password_validation.validate_password])
+
+    def save(self, user):
+        user.set_password(self.data["new_password"])
```

Comment:
self.data represents the given data. You should always use the `validated_data`  to get the validated data.